protocols/kad/src/kbucket/key.rs

```diff
@@ -92,10 +91,10 @@ impl<T> Into<KeyBytes> for Key<T> {
     }
 }
 
-impl From<Multihash> for Key<Multihash> {
```

Comment:
Is there a specific reason for replacing this instance with `From<Vec<u8>>`?

Reply:
The reason this is removed that it is not really possible to have a `Key<Multihash>` implementation. The problem is that `Key` needs to implement `Borrow<[u8]>`. But you cannot implement `Borrow<[u8]>` for Multihash as it doesn't own a continuous sequence of bytes containing the full multihash.

This relates to that comment https://github.com/libp2p/rust-libp2p/pull/1792#pullrequestreview-523440886 where I removed the `Borrow` to `Into<Vec<u8>>` in order to be able to implement `Key<Multihash>`.

Reply:
> The problem is that Key needs to implement Borrow<[u8]>.

Not quite, the type itself does not. The public constructors `Key::new` and `KeyBytes::new` need this constraint, but I think the `From<Multihash>` instance can bypass these (possibly extracting code from these public constructors to internal ones that can be used in both places), as it can ensure that the byte representation of the multihash that is used to create the `Key` is consistent with the original `Multihash` w.r.t. equality. Would that not work? I guess there is no reason not to also have a `From<Vec<u8>>` instance but it would be preferable not to settle for a `Key<Vec<u8>>` when one wants a `Key<Multihash>`.

Reply:
You are absolutely right. Sorry that I missed that. Thanks for the detailed review and helping me.